dev-scripts/run.py

```
@@ -359,12 +359,33 @@ def check_dependencies(args):
-    # Ensure directory exists for sqlite
+    # Ensure directory exists for sqlite.
+    # For postgres/mysql, ensure the database itself exists — `dbmate
+    # up` used to do this implicitly, but `ncps migrate up` connects
+    # to an existing database and won't create it. We use `dbmate
+    # create` (idempotent-ish: errors if it already exists, which we
+    # ignore) to keep the behavior compatible.
     if db_url.startswith("sqlite:"):
         path = db_url.replace("sqlite:", "")
         os.makedirs(os.path.dirname(path), exist_ok=True)
+    else:
+        subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)